File: candado/solutions/correct/nlehmann.cpp
#include <cstdio>
#include <vector>
#include <algorithm>

using namespace std;

void read_combination(vector<int> &vec) {
  for (int i = 0; i < 4; ++i) {
    scanf("%d", &vec[i]);
  }
}

int main() {
  vector<int> start(4);
  vector<int> end(4);
  read_combination(start);
  read_combination(end);

  int s = 0;
  for (int i = 0; i < 4; ++i) {
    if (start[i] > end[i]) {
      s += min(start[i] - end[i], 10 + end[i] - start[i]);
    } else {
      s += min(end[i] - start[i], 10 + start[i] - end[i]);
    }
  }
  printf("%d\n", s);

  return 0;
}

File: candado/solutions/correct/nlehmann.java
import java.util.*;
import java.io.*;

class candado {
    static public void main(String argv[]) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

        int start[] = readCombination(new StringTokenizer(in.readLine()));
        int end[] = readCombination(new StringTokenizer(in.readLine()));

        int s = 0;
        for (int i = 0; i < 4; ++i) {
            s += Math.min((10 + start[i] - end[i]) % 10, (10 + end[i] - start[i]) % 10);
        }
        System.out.println(s);
    }

    static int[] readCombination(StringTokenizer st) {
        int arr[] = new int[4];
        for (int i = 0; i < 4; ++i) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        return arr;
    }
}
